### PR TITLE
Add flaky test issue template

### PR DESCRIPTION
**Pull request by andrewvc**
This PR attempts to define an issue template for reporting flakey tests.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/flaky-test.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+---
+name: Flaky Test
+about: Report a flakey test (one that doesn't pass consistently)
+
+---
+
+## Flakey Test
```
Comment (andrewkroh) [inline .github/ISSUE_TEMPLATE/flaky-test.md]: s/Flakey/Flaky/

Comment (andrewvc) [reply .github/ISSUE_TEMPLATE/flaky-test.md]: Hey, turns out they're both [valid](https://www.merriam-webster.com/dictionary/flaky) spellings, but I'll drop the e here for consistency. Good find.

**Comment by andrewvc** [comment]
@andrewkroh made spelling of flaky consistent

**Comment by andrewvc** [comment]
This may seem like overkill, but I copied most of Kibana's PR guidelines and included them here to address the feedback that we add this to the dev guide. We didn't really have a great place to put this info. So, I created one.

I can do something more minimal if need be, but I think the kibana guide is pretty non-controversial. Alternatively we could just add a paragraph to https://www.elastic.co/guide/en/beats/devguide/current/beats-contributing.html#contribution-steps

However, I think this guide is a good idea, especially as more people work on beats.

File: docs/devguide/pull-request-guidelines.asciidoc [hunk 2]
```diff
@@ -0,0 +1,129 @@
+[[pr-review]]
```
Comment (ruflin) [inline docs/devguide/pull-request-guidelines.asciidoc]: Let's move this to a separate PR for discussion and focus this PR only on the template.

File: .github/ISSUE_TEMPLATE/flaky-test.md (hunk 1 above)
Comment (andrewkroh) [reply .github/ISSUE_TEMPLATE/flaky-test.md]: Good to know. I had no idea flakey was also valid, but it stood out as inconsistent.

**Comment by andrewvc** [comment]
@ruflin I've moved the docs changes to https://github.com/elastic/beats/pull/8968

**Comment by andrewvc** [comment]
FYI, I've actually had a chance to use this template in https://github.com/elastic/beats/issues/8966 for those who want to see it in action.

File: docs/devguide/pull-request-guidelines.asciidoc (hunk 2 above)
Comment (andrewvc) [reply docs/devguide/pull-request-guidelines.asciidoc]: Done

**Comment by andrewvc** [comment]
@ruflin @andrewkroh are there any objects to merging this as is?
